AIO/src/org/aio/activities/quests/TheRestlessGhost.java

```diff
@@ -0,0 +1,190 @@
+            depositAllBanking.run();
+        } else if (getTabs().getOpen() != Tab.INVENTORY) {
+            getTabs().open(Tab.INVENTORY);
+        } else {
+            switch (getProgress()) {
+                case 0:
+                    talkToAereck();
+                    break;
+                case 1:
+                    talkToUrhney();
+                    break;
+                case 2:
+                    talkToGhost();
+                    break;
+                case 3:
+                    searchAlter();
+                    break;
+                case 4:
+                    placeSkull();
+                    break;
+                case 5:
+                    // Make sure we are not in the cut scene
+                    if (getWidgets().get(277, 2).isVisible()) {
+                        log("Quest is complete");
+                        isComplete = true;
+                    }
+                    break;
+                default:
+                    log("Unknown progress config value: " + getProgress());
+                    setFailed();
+                    break;
+            }
+        }
+    }
+
+    private void placeSkull() throws InterruptedException {
+        RS2Object coffin = getObjects().closest("Coffin");
+
+        if (!COFFIN.contains(myPosition())) {
+            getWalking().webWalk(COFFIN);
+        } else if(coffin != null && coffin.hasAction("Close") && getInventory().contains("Ghost's skull")) {
+            useSkull();
+        } else if (coffin != null && coffin.hasAction("Open")) {
+            openCoffin();
+        }
+    }
+
+    private void useSkull() throws InterruptedException {
+        if(getInventory().isItemSelected()) {
+            if(getObjects().closest("Coffin").interact("Use")){
+                Sleep.sleepUntil(() -> getProgress() != 4, 15000, 1500);
+            }
+        } else {
+            if(getInventory().interact("Use", "Ghost's skull")) {
+                Sleep.sleepUntil(() -> getInventory().isItemSelected(), 5000);
+            }
+        }
+    }
+
+    private void talkToGhost() throws InterruptedException {
+        NPC ghost = getNpcs().closest("Restless ghost");
+
+        if (!COFFIN.contains(myPosition())) {
+            getWalking().webWalk(COFFIN);
+        } else if (ghost == null) {
+            openCoffin();
+        } else if (canTalkToGhost() && (!getDialogues().inDialogue() || !myPlayer().isInteracting(ghost))) {
+            if (ghost.interact("Talk-to")) {
+                Sleep.sleepUntil(() -> getDialogues().inDialogue() && myPlayer().isInteracting(ghost), 5000);
+            }
+        } else {
+            getDialogues().completeDialogue(GHOST_OPTIONS);
+        }
+    }
+
+    private void openCoffin() throws InterruptedException {
+        RS2Object coffin = getObjects().closest("Coffin");
+
+        if (coffin != null && (coffin.interact("Search") || coffin.interact("Open"))) {
```

Comment:
Instead of this or condition, you can just do: `interact("Search", "Open")`

Reply:
Great! Learning so much.